src/Microsoft.DotNet.Build.Tasks/GetDailyBuildNumber.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using LibGit2Sharp;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class GetDailyBuildNumber : Task
+    {
+        [Required]
+        public string VersionPropsFile { get; set; }
+
+        [Output]
+        public string DailyBuildNumber { get; set; }
+
+        public override bool Execute()
+        {
+            Log.LogMessage("Starting GetDailyBuildNumber");
+
+            XDocument versionProps = XDocument.Load(VersionPropsFile);
+            var defaultNS = versionProps.Root.GetDefaultNamespace();
+
+            var dailyBuildNode =
+                (from el in versionProps.Descendants(defaultNS + "DailyBuildNumber")
+                 select el).FirstOrDefault();
+
+            Debug.Assert(dailyBuildNode != null, "dailyBuildNode shouldn't be null");
+            var incDailyBuildNum = int.Parse(dailyBuildNode.Value) + 1;
```

Comment:
If the data is malformed how do you suggest we recover?  I'd prefer to fail-fast in the event of malformed data so people know it's broken.  I will add a comment to the props file indicating that the value must be numeric.